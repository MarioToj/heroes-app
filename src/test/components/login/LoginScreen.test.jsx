
import { MemoryRouter, Routes, Route } from "react-router"
import { mount } from "enzyme"
import { AuthContext } from "../../../auth/authContext"
import { LoginScreen } from "../../../components/login/LoginScreen"
import { types } from "../../../types/types"

const mockNavigate = jest.fn()

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate,
}))

describe('Pruebas en <LoginScreen/>', () => {
    
    const contexValue = {
        dispatch: jest.fn(),
        user:{

            logged: false

        }

    }

    const wrapper = mount(

        <AuthContext.Provider value={ contexValue } >

            <MemoryRouter initialEntries={['/login']} >
                <Routes>
                    <Route path='/login' element={<LoginScreen/>}/>
                    
                </Routes>
            </MemoryRouter>

        </AuthContext.Provider>

    )

        test('debe de hacer match con el snapshot ', () => {
            
            expect( wrapper ).toMatchSnapshot()

        })

        test('debe de realizar el dispatch ', () => {
            
            const handleClick = wrapper.find('button').prop('onClick')
            handleClick()

            expect( contexValue.dispatch ).toHaveBeenCalledWith({
                type: types.login,
                payload: { name: "Mario" }
            }) 

            expect( mockNavigate ).toHaveBeenCalledWith( '/marvel', { replace: true } ) 

        })
        

})