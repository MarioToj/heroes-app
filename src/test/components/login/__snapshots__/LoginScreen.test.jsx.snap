// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen/> debe de hacer match con el snapshot  1`] = `""`;

exports[`Pruebas en <NavBar/> debe de hacer match con el snapshot  1`] = `""`;
